subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/fingerprint/ConfigurationCacheFingerprintController.kt

```diff
@@ -51,13 +54,15 @@ class ConfigurationCacheFingerprintController internal constructor(
     private val taskInputsListeners: TaskInputsListeners,
     private val valueSourceProviderFactory: ValueSourceProviderFactory,
     private val fileSystemAccess: FileSystemAccess,
-    private val fileCollectionFingerprinter: AbsolutePathFileCollectionFingerprinter,
+    private val fingerprinterRegistry: FileCollectionFingerprinterRegistry,
     private val buildCommencedTimeProvider: BuildCommencedTimeProvider,
     private val listenerManager: ListenerManager,
     private val buildTreeListenerManager: BuildTreeListenerManager,
     private val fileCollectionFactory: FileCollectionFactory,
     private val directoryFileTreeFactory: DirectoryFileTreeFactory
 ) : Stoppable {
+    private
+    val fileCollectionFingerprinter = fingerprinterRegistry.getFingerprinter(DefaultFingerprinterSpec.from(AbsolutePathInputNormalizer::class.java, DirectorySensitivity.DEFAULT))
```

Comment:
It would be good to avoid injecting the whole registry here. I think it should be possible to create a service-provider method that passes in the single `FileCollectionFingerprinter` here instead.

Reply:
I'm not sure if what I did was what you were expecting or if there is some obscure feature of `ServiceRegistry` you wanted me to use.  Basically, the issue here is that looking up just an `AbsolutePathFileCollectionFingerprinter` doesn't work, because there are multiple registered objects of that type (`FileCollectionFingerprinterRegistry` grabs all of those and stuffs them in the registry).  We might want to think about registering the instances in the fingerprinter registry differently so we only have to provide service methods for the fingerprinter(s) that we need to get explicitly.  

If you had something else in mind, let me know.

Reply:
OK. Let's just leave this as it is now.